framework/src/controller/applicationState.js

```diff
@@ -35,38 +36,29 @@ const semver = require('semver');
  * @requires crypto
  * @requires os
  * @requires semver
- * @param {Object} options - System options
+ * @param {Object} initialState - Initial state of the application
  * @param {Object} logger
  */
-class System {
-	constructor(config, logger, storage) {
+class ApplicationState {
+	constructor(initialState, logger, channel) {
 		this.logger = logger;
-		this.storage = storage;
-		this.headers = {
+		this.state = {
 			os: os.platform() + os.release(),
-			version: config.version,
-			wsPort: config.wsPort,
-			httpPort: config.httpPort,
-			minVersion: config.minVersion,
-			protocolVersion: config.protocolVersion,
+			version: initialState.version,
+			wsPort: initialState.wsPort,
+			httpPort: initialState.httpPort,
+			minVersion: initialState.minVersion,
+			protocolVersion: initialState.protocolVersion,
 			height: 1,
-			nethash: config.nethash,
-			broadhash: config.nethash,
-			nonce: config.nonce,
+			nethash: initialState.nethash,
+			broadhash: initialState.nethash,
+			nonce: initialState.nonce,
 		};
+		this.channel = channel;
 	}
 
-	/**
-	 * Checks nethash (network) compatibility.
-	 *
-	 * @param {string} nethash
-	 * @returns {boolean}
-	 */
-	networkCompatible(nethash) {
-		if (!nethash) {
-			return false;
-		}
-		return this.headers.nethash === nethash;
+	getState() {
```

Comment:
`setState` is missing. 

`setState` can only update the state if a module has a sufficient authorization i.e as described in issue #3036 
> The following variables can only be assigned once after the application starts and only by application controller:
os
nonce
version
minVersion
protocolVersion
nethash
The following variables can only be modified by the chain module:
height
broadhash

Reply:
As discussed in the offline meeting, to do this, the corresponding channel authorization needs to be implemented, for application to be able to verify a channel of which module is invoking the `setState`.

Reply:
The state is initiated in the constructor passing the config variables as argument (`initialState`). What we need to moderate is which module is able to call the update action.